src/Aspire.Hosting.Azure/api/Aspire.Hosting.Azure.cs

```diff
@@ -252,8 +287,93 @@ public ExistingAzureResourceAnnotation(object name, object? resourceGroup = null
         public object? ResourceGroup { get { throw null; } }
     }
 
+    public partial interface IAddRoleAssignmentsContext
+    {
+        AzureResourceInfrastructure Infrastructure { get; }
+
+        global::Azure.Provisioning.BicepValue<System.Guid> PrincipalId { get; }
+
+        global::Azure.Provisioning.BicepValue<string> PrincipalName { get; }
+
+        global::Azure.Provisioning.BicepValue<global::Azure.Provisioning.Authorization.RoleManagementPrincipalType> PrincipalType { get; }
+
+        System.Collections.Generic.IEnumerable<RoleDefinition> Roles { get; }
+    }
+
+    public partial interface IAppIdentityResource
+    {
+        BicepOutputReference ClientId { get; }
+
+        BicepOutputReference Id { get; }
+
+        BicepOutputReference PrincipalId { get; }
+
+        BicepOutputReference PrincipalName { get; }
+    }
+
+    public partial interface IKeyVaultResource : ApplicationModel.IResource, ApplicationModel.IAzureResource
+    {
+        BicepOutputReference IdOutputReference { get; }
+
+        BicepOutputReference NameOutputReference { get; }
+
+        System.Func<string, System.Threading.CancellationToken, System.Threading.Tasks.Task<string?>>? SecretResolver { get; set; }
```

Comment:
I plan to change this to take the IKeyVaultSecretReference as the first argument.